From c7d2cb62412fcac8a0a1f1cf26e481427591098e Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Tue, 12 May 2020 03:51:40 -0700
Subject: [PATCH 13/14] Revert "mm/hmm: make CONFIG_DEVICE_PRIVATE into a
 select"

This reverts commit d2c63df2242ee20deda58506eb573356f4f6f2ac.

This was dropped in the latest hmm branch.

Link: https://lore.kernel.org/lkml/20200511135257.GT19158@mellanox.com/
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 arch/powerpc/Kconfig            | 2 +-
 drivers/gpu/drm/nouveau/Kconfig | 2 +-
 lib/Kconfig.debug               | 2 +-
 mm/Kconfig                      | 7 ++++++-
 4 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index ab40b88eb323..bed534711128 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -455,7 +455,7 @@ config PPC_TRANSACTIONAL_MEM
 config PPC_UV
 	bool "Ultravisor support"
 	depends on KVM_BOOK3S_HV_POSSIBLE
-	select DEVICE_PRIVATE
+	depends on DEVICE_PRIVATE
 	default n
 	help
 	  This option paravirtualizes the kernel to run in POWER platforms that
diff --git a/drivers/gpu/drm/nouveau/Kconfig b/drivers/gpu/drm/nouveau/Kconfig
index af5793f3e7c2..d6e4ae1ef705 100644
--- a/drivers/gpu/drm/nouveau/Kconfig
+++ b/drivers/gpu/drm/nouveau/Kconfig
@@ -86,10 +86,10 @@ config DRM_NOUVEAU_BACKLIGHT
 
 config DRM_NOUVEAU_SVM
 	bool "(EXPERIMENTAL) Enable SVM (Shared Virtual Memory) support"
+	depends on DEVICE_PRIVATE
 	depends on DRM_NOUVEAU
 	depends on MMU
 	depends on STAGING
-	select DEVICE_PRIVATE
 	select HMM_MIRROR
 	select MMU_NOTIFIER
 	default n
diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 2015a77b4d91..f7d80d6241b8 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -2273,7 +2273,7 @@ config TEST_MEMINIT
 config TEST_HMM
 	tristate "Test HMM (Heterogeneous Memory Management)"
 	depends on TRANSPARENT_HUGEPAGE
-	select DEVICE_PRIVATE
+	depends on DEVICE_PRIVATE
 	select HMM_MIRROR
 	select MMU_NOTIFIER
 	help
diff --git a/mm/Kconfig b/mm/Kconfig
index 761a54d1c32c..db626b8d4fdb 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -805,10 +805,15 @@ config HMM_MIRROR
 	depends on MMU
 
 config DEVICE_PRIVATE
-	bool
+	bool "Unaddressable device memory (GPU memory, ...)"
 	depends on ZONE_DEVICE
 	select DEV_PAGEMAP_OPS
 
+	help
+	  Allows creation of struct pages to represent unaddressable device
+	  memory; i.e., memory that is only accessible from the device (or
+	  group of devices). You likely also want to select HMM_MIRROR.
+
 config FRAME_VECTOR
 	bool
 
-- 
2.26.2

